From: wuyifeng <wuyifeng@nd.com.cn>
Date: Fri, 15 Dec 2022 10:07:07 +0800
Subject: Add Ctrl+Alt+D hotkey to save all open tabs as bookmarks

Fix version: 1.2.1
Add Ctrl+Alt+D hotkey to save all open tabs as bookmarks when the user signs in to Promethean.
Relate to AP9-11086

---
 chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java | 9 +++++++++
 1 file changed, 9 insertions(+), 0 deletion(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java b/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/KeyboardShortcuts.java
@@ -182,6 +182,8 @@ public class KeyboardShortcuts {
         if (isLogin) {
             addShortcut(context, webpageShortcutGroup, R.string.keyboard_shortcut_bookmark_page,
                 KeyEvent.KEYCODE_D, KeyEvent.META_CTRL_ON);
+            addShortcut(context, webpageShortcutGroup, R.string.keyboard_shortcut_bookmark_page,
+                KeyEvent.KEYCODE_D, ctrlShift);
         }
         shortcutGroups.add(webpageShortcutGroup);

@@ -379,6 +381,13 @@ public class KeyboardShortcuts {
                     menuOrKeyboardActionController.onMenuOrKeyboardAction(
                         R.id.bookmark_this_page_id, false);
                     return true;
+                case CTRL | SHIFT | KeyEvent.KEYCODE_D:
+                    if (!isLogin) {
+                        return false;
+                    }
+                    menuOrKeyboardActionController.onMenuOrKeyboardAction(
+                        R.id.bookmark_all_tabs_menu_id, false);
+                    return true;
             }
         }

--
2.34.1